nucypher/blockchain/eth/actors.py

```diff
@@ -1062,23 +1054,26 @@ def generate_policy_parameters(self,
             raise ValueError("Policy end time must be specified as 'expiration' or 'duration_periods', got neither.")
 
         # Merge injected and default params.
-        first_period_reward = first_period_reward or self.first_period_reward
         rate = rate or self.rate
         duration_periods = duration_periods or self.duration_periods
 
         # Calculate duration in periods and expiration datetime
-        if expiration:
-            duration_periods = calculate_period_duration(future_time=expiration,
-                                                         seconds_per_period=self.economics.seconds_per_period)
-        else:
-            duration_periods = duration_periods or self.duration_periods
+        if duration_periods:
+            # Duration equals one period means that expiration date is the last second of the current period
             expiration = datetime_at_period(self.staking_agent.get_current_period() + duration_periods,
-                                            seconds_per_period=self.economics.seconds_per_period)
+                                            seconds_per_period=self.economics.seconds_per_period,
+                                            start_of_period=True)
```

Comment:
The comment says "expiration date is the last second of the current period", but here we pass `start_of_period=True` - what am I missing?

Reply:
We take start of the next period and subtract 1 second (next line)

Reply:
I'm sorry, maybe I'm being thick here, but I still don't get it.  How does subtracting one second from the `MayaDT` get us to the last second of the current period?  And how does `start_of_period=True` relate to that?

Reply:
maybe comment is misleading, current period will be only for case when `duration_periods==1`, sort of example.
In that case `start_of_period=True` will return zero/first second of the next period and subtracting 1 second we will come to the last second of the previous period. 
If `duration_periods==2` then `start_of_period=True` will be period after next period and `expiration` after subtracting will be last second of the next period
`start_of_period=True` is sort of rounding, returns the beggining of calculated period